### DISCUSSION TITLE
Is there a recommended way of creating mock functions using cmake-init?

### DISCUSSION THREAD

**Discussion by Kyr27** [friendlyanon/cmake-init#123] [Q&A] [answered]
Hello! I’m currently testing my C library using the [Unity testing framework](https://github.com/ThrowTheSwitch/Unity) and cmake-init, and everything has gone smoothly so far. However, I’ve reached a point where I need to mock some Windows-specific functions, such as ```PostMessage```, to get my tests to pass successfully.

Given how amazing this CMake initializer is, I have a question: Is there a recommended way to mock functions using cmake-init? If possible, could you provide an example?

Thank you!

**Comment by friendlyanon** [comment, marked as answer]
You don't "mock functions using cmake-init". You can just use regular old dependency injection and not hardcode the use of `PostMessage`, but that's just one way of dealing with making software testable.

**Reply by Kyr27** [reply]
Thank you for your fast response. 
I appreciate the suggestion about dependency injection. I’m already using it in my setup, so I should have been clearer in my question. What I meant by "mock functions using cmake-init" was to ask if cmake-init has a recommended approach or an opinion for handling function mocking, given how opinionated it is in other areas.

I’ll take your answer as a "no" and continue with my current dependency injection setup. Thanks once again.
